test/integration/apiserver/anonymous/anonymous_test.go

```diff
@@ -0,0 +1,332 @@
+		ObjectMeta: metav1.ObjectMeta{Name: defaultRBACRoleName},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:         []string{"list"},
+				Resources:     []string{"pods"},
+				APIGroups:     []string{""},
+				ResourceNames: []string{},
+			},
+		},
+	}
+	defaultRoleBinding = &rbacv1.RoleBinding{
+		TypeMeta:   metav1.TypeMeta{APIVersion: "rbac.authorization.k8s.io/v1", Kind: "RoleBinding"},
+		ObjectMeta: metav1.ObjectMeta{Name: defaultRBACRoleBindingName},
+		Subjects: []rbacv1.Subject{
+			{
+				APIGroup: rbac.GroupName,
+				Kind:     rbacv1.UserKind,
+				Name:     "system:anonymous",
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbac.GroupName,
+			Kind:     "Role",
+			Name:     defaultRBACRoleName,
+		},
+	}
+)
+
+func TestStructuredAuthenticationConfig(t *testing.T) {
+	t.Log("Testing anonymous authenticator with authentication config")
+
+	testCases := []struct {
+		desc            string
+		featureGates    map[featuregate.Feature]bool
+		authConfig      string
+		additionalFlags []string
+		assertErrFn     func(t *testing.T, errorToCheck error)
+		assertFn        func(t *testing.T, server kubeapiserverapptesting.TestServer)
+	}{
+		{
+			desc: "StructuredAuthenticationConfiguration disabled",
+			featureGates: map[featuregate.Feature]bool{
+				features.StructuredAuthenticationConfiguration: false,
+			},
+			authConfig: `
+apiVersion: apiserver.config.k8s.io/v1beta1
+kind: AuthenticationConfiguration
+anonymous:
+  enabled: true
+  conditions:
+  - path: "/livez"
+`,
+			assertErrFn: func(t *testing.T, errorToCheck error) {
+				assert.EqualError(t, errorToCheck, "failed to validate ServerRunOptions: set --feature-gates=StructuredAuthenticationConfiguration=true to use authentication-config file")
+			},
+		},
+		{
+			desc: "AnonymousAuthConfigurableEndpoints disabled",
+			featureGates: map[featuregate.Feature]bool{
+				features.StructuredAuthenticationConfiguration: true,
+				features.AnonymousAuthConfigurableEndpoints:    false,
+			},
+			authConfig: `
+apiVersion: apiserver.config.k8s.io/v1beta1
+kind: AuthenticationConfiguration
+anonymous:
+  enabled: true
+  conditions:
+  - path: "/livez"
+`,
+			assertErrFn: func(t *testing.T, errorToCheck error) {
+				assert.EqualError(t, errorToCheck, "anonymous: Forbidden: anonymous is not supported when when AnonymousAuthConfigurableEnpoints feature gate is disabled")
+			},
+		},
+		{
+			desc: "invalid flags",
+			featureGates: map[featuregate.Feature]bool{
+				features.StructuredAuthenticationConfiguration: true,
+				features.AnonymousAuthConfigurableEndpoints:    true,
+			},
+			authConfig: `
+apiVersion: apiserver.config.k8s.io/v1beta1
+kind: AuthenticationConfiguration
+anonymous:
+  enabled: true
+  conditions:
+  - path: "/livez"
+`,
+			assertErrFn: func(t *testing.T, errorToCheck error) {
+				assert.EqualError(t, errorToCheck, "anonymous: Forbidden: --anonynous-auth flag cannot be set when anonymous field is configured in authentication configuration file")
```

Comment:
cannot be set to "true", correct? because the default is "false".
might be worth clarifying that point in the error message.

Reply:
the flag cannot be explicitly set when the anonymous field is configured in the authentication config file.

Reply:
yes, my point was to clarify in the lines of:
`assert.EqualError(t, errorToCheck, "anonymous: Forbidden: --anonynous-auth flag cannot be set to "true" when anonymous field is configured in authentication configuration file"`